### PR TITLE
Docstrings for every module

### PR DESCRIPTION
**Pull request by nils-herrmann**
The following changes were made:
- Docstring for files and functions (with exception of tests)
- Imports at top of files (with exception of importing `Scientist` class)
- Explicit exceptions
- Removal of unnecessary parenthesis

### CODE REVIEW

File: sosia/classes/original.py [hunk 1]
```diff
@@ -35,10 +33,7 @@ def search_group(self):
         -----
         Property is initiated via .define_search_group().
         """
-        try:
-            return self._search_group
-        except AttributeError:
-            return None
+        return self._search_group
```
Comment (nils-herrmann) [inline sosia/classes/original.py:36]: Simplification by assigning `None` to self._search_group as default (see below)

File: sosia/classes/original.py [hunk 2]
```diff
@@ -18,10 +19,7 @@ def matches(self):
         -----
         Property is initiated via .find_matches().
         """
-        try:
-            return self._matches
-        except AttributeError:
-            return None
+        return self._matches
```
Comment (nils-herrmann) [inline sosia/classes/original.py:22]: Simplification by assigning `None` to self._matches as default (see below)

File: sosia/processing/querying.py [hunk 3]
```diff
@@ -271,5 +273,5 @@ def valid_results(res):
     try:
         _ = [p for p in res if p.subtype == "ar" and int(p.coverDate[:4])]
         return True
-    except:
+    except (AttributeError, TypeError, ValueError):
```
Comment (nils-herrmann) [inline sosia/processing/querying.py:276]: Should catch all the possible exceptions:
- AttributeError: `p.subtype` or `p.coverDate` might not exist.
- TypeError: `res` might not be a list or iterable, or `coverDate` might not be a string.
- ValueError: `int(p.coverDate[:4])` might fail if `coverDate[:4]`is not a valid integer string.

File: sosia/processing/filtering.py [hunk 4]
```diff
@@ -1,3 +1,10 @@
+"""
+Module with a function for filtering authors based on restrictions in the number of 
+publications in different periods.
+"""
+
+from itertools import product
+from pandas import DataFrame
```
Comment (Michael-E-Rose) [inline sosia/processing/filtering.py:7]: There needs to be a newline here

File: sosia/processing/finding.py [hunk 5]
```diff
@@ -1,3 +1,8 @@
+"""
+Module with functions for finding matches within a search group based on various criteria
```
Comment (Michael-E-Rose) [inline sosia/processing/finding.py:2]: Docstring should start on first line.

File: sosia/establishing/fields_sources.py [hunk 6]
```diff
@@ -1,3 +1,8 @@
+"""
```
Comment (Michael-E-Rose) [inline sosia/establishing/fields_sources.py]: Docstring should start on first line.

File: sosia/establishing/__init__.py [hunk 7]
```diff
@@ -1,3 +1,7 @@
+"""
```
Comment (Michael-E-Rose) [inline sosia/establishing/__init__.py]: Docstring should start on first line.

File: sosia/classes/__init__.py [hunk 8]
```diff
@@ -1,2 +1,7 @@
+"""
```
Comment (Michael-E-Rose) [inline sosia/classes/__init__.py]: Docstring should start on first line.

File: sosia/processing/finding.py [hunk 9]
```diff
@@ -1,3 +1,8 @@
+"""
```
Comment (Michael-E-Rose) [inline sosia/processing/finding.py]: Docstring should start on first line.

File: sosia/processing/filtering.py [hunk 10]
```diff
@@ -1,3 +1,10 @@
+"""
```
Comment (Michael-E-Rose) [inline sosia/processing/filtering.py]: Docstring should start on first line.

**Review by Michael-E-Rose — CHANGES_REQUESTED** [review]
Let docstrings always start on first line, and separate python base packages from third party packages.

**Review by Michael-E-Rose — APPROVED** [review]

**Pull request merged**